database/src/main/resources/db/migration/V1_169__change_address_post_code.sql

```diff
@@ -0,0 +1,65 @@
+ALTER TYPE address
+    ADD ATTRIBUTE "foreign_postal_code" VARCHAR(10);
+
+-- Create backup for "transport-service" table
+CREATE TABLE "transport-service_backup" AS TABLE "transport-service";
+
+CREATE OR REPLACE FUNCTION convert_transport_service_address_postalcodes()
+RETURNS VOID AS $$
+BEGIN
+    -- Update service address
+    UPDATE "transport-service" as ts
+       SET ("contact-address"."foreign_postal_code") =
+            (SELECT (service."contact-address")."postal_code" as postalcode
+               FROM "transport-service" service
+              WHERE service.id = ts.id);
+
+    -- update rental pick up addressess
+    UPDATE "transport-service" as ts
+    SET (rentals."pick-up-locations") =
+            (SELECT array_agg(ROW(pa."pick-up-name",
+                                  pa."pick-up-type",
+                                  pa."service-hours",
+                                  pa."service-exceptions",
+                                  ((pa."pick-up-address").street,
+                                   (pa."pick-up-address").postal_code,
+                                   (pa."pick-up-address").post_office,
+                                   (pa."pick-up-address").postal_code
+                                  )::address,
+                                  pa."service-hours-info")::pick_up_location)
+               FROM "transport-service" service,
+                    lateral unnest((service.rentals)."pick-up-locations") pa
+              WHERE service.type = 'rentals'
+                AND ts.id = service.id
+              GROUP BY service.id);
+END;
+$$ LANGUAGE plpgsql;
+
+SELECT convert_transport_service_address_postalcodes();
+
+CREATE OR REPLACE FUNCTION convert_transport_operator_address_postalcodes()
+RETURNS VOID
+AS $$
+BEGIN
+    UPDATE "transport-operator" as "t-operator"
+       SET ("visiting-address"."foreign_postal_code") =
+           (SELECT (top."visiting-address")."postal_code" as postalcode
+             FROM "transport-operator" top
+            WHERE top.id = "t-operator".id);
+
+    UPDATE "transport-operator" as "t-operator"
+       SET ("billing-address"."foreign_postal_code") =
+           (SELECT (top."billing-address")."postal_code" as postalcode
+              FROM "transport-operator" top
+            WHERE top.id = "t-operator".id);
+
+END;
+$$ LANGUAGE plpgsql;
+
+SELECT convert_transport_operator_address_postalcodes();
+
+ALTER TYPE address
+    DROP ATTRIBUTE "postal_code";
```

Comment:
Doesn't dropping composite type's attributes cause trouble when there's already data in db created using that template?
E.g. in this file `rentals."pick-up-locations"` migration puts postal_code in the middle and foreign_postal_code in the end. Won't dropping postal_code lead to data in the db array not matching to the type anymore?

Reply:
Dropping column from composite type seems to be ok. It works fine. Though I needed to update our test data sql files.